snowflake_blueprints/upload_file.py

```diff
@@ -161,14 +192,24 @@ def execute_copyinto_command(db_connection, table_name, results_dict):
 def upload_data_with_put(source_full_path,
                          table_name,
                          insert_method,
-                         db_connection):
+                         db_connection,
+                         snowflake_data_types):
     """
     Upload data by PUTing the file(s) in Snowflake temporary storage and using COPY INTO to get them into the table.
     """
     parquet_path = convert_to_parquet(source_full_path, table_name)
     print('Attempting upload with put method')
     snowflake_results = {"put": [], "copy": [], "drop": []}
-    if insert_method == 'replace':
+    # Adding logic to create table with types if options is provided
+    # This will supercede the insert method
+    if snowflake_data_types:
+        create_table_with_types(
```

Comment:
This ordering seems off to me. Currently it says, no matter if they replace or append, you're going to create the table and insert the data.

Instead, it should be a branch off the replace and append logic. If replace AND snowflake_data_types, create_table_with_types, otherwise create_table. If append AND snowflake_data_types, create_table_with_types, otherwise create_table.